packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +292,99 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def dns_managed?
+      @scenario_answers['foreman_proxy']['dns'] &&
+        @scenario_answers['foreman_proxy']['dns_managed'] &&
+        @scenario_answers['foreman_proxy']['dns_provider'] == 'nsupdate'
+    end
+
+    def should_update_dns?
+      dns_managed? && !@options[:skip_dns]
+    end
+
+    def answers_dns_values
+      {
+        nameserver_ip: @scenario_answers['foreman_proxy']['dns_server'],
+        zone: @scenario_answers['foreman_proxy']['dns_zone'],
+        key_file: @scenario_answers['foreman_proxy']['keyfile'],
+        reverse_zones: [@scenario_answers['foreman_proxy']['dns_reverse']].flatten
```

Comment:
The value may be `nil` and you don't want to end up with `[nil]`
```suggestion
        reverse_zones: [@scenario_answers['foreman_proxy']['dns_reverse']].flatten.compact
```